tools/myst/host/exec_linux.c

```diff
@@ -309,6 +309,15 @@ void cleanup_alt_stack()
 
 static void _sigaction_handler(int sig, siginfo_t* si, void* context)
 {
+    /* Unblock signal */
+    sigset_t mask;
+    sigemptyset(&mask);
+    sigaddset(&mask, sig);
+
+    int ret = pthread_sigmask(SIG_UNBLOCK, &mask, NULL);
```

Comment:
Not sure I get this correctly: if we are here, that means the signal has already been delivered to *this* thread, that implied the mask is off for this thread. Why unmask again?

Reply:
Is this for the 2nd firing of SIGSEGV? We are supposed to turn on the mask for SIGSEGV when handing SIGSEGV, and turn it off when finished handling.  We need to understand why the mask stays on after the handler is done.

Reply:
We are unblocking the signal for this thread. So assuming the signal handler doesn't return, and there are more instances of this signal; we are able to deliver those signals.

Reply:
I am still not sure why the signal handler didn't return from the 1st SIGSEGV. But if we want nested handling of SIGSEGV, we could use SA_NODEFER while calling `sigaction`

Reply:
From dotnet team engineer:

>  the SIGSEGV handler never returns. We always start the exception handling directly from the handler. To execute the catch handler, we set the CPU context using our the RtlRestoreContext function.

Reply:
I am not sure about the "never returns". We only have one thread in the test case, if the signal handler never returns, the test should hang, right?

It should be easy to find out by debugging `myst_handle_host_signal`

Reply:
Thanks @vtikoo  for verifying this. The exception handling on Linux indeed doesn't return.